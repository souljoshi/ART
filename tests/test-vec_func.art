void donothing(vec e)
{
    e = <2.0, 2.0>;
}

void zero_vec(vec &v)
{
    v = <0.0, 0.0>;
}

vec twice(vec &v)
{
    return 2 * v;
}

void print_vec(vec v)
{
    printc('[');
    printf(v[0]);
    printc(',');
    printf(v[1]);
    printc(']');
    printc('\n');
}

int main()
{
    vec v;

    v = <2.0, 5.0>;
    print_vec(v);
    donothing(v);
    print_vec(v);
    print_vec(twice(v));
    zero_vec(v);
    print_vec(v);
}