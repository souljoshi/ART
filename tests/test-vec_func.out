[2.0,2.5]
[2.0,2.5]
[4.0,5.0]
[0.0,0.0]